crypto/s2n_hmac.c

```diff
@@ -252,14 +264,19 @@ int s2n_hmac_update(struct s2n_hmac_state *state, const void *in, uint32_t size)
      * input. On some platforms, including Intel, the operation can take a
      * smaller number of cycles if the input is "small".
      */
-    state->currently_in_hash_block += (4294949760 + size) % state->hash_block_size;
+    ENSURE_POSIX(state->hash_block_size != 0, S2N_ERR_INTEGER_OVERFLOW);
```

Comment:
This looks a bit odd to me...

Should this be an overflow error or just a precondition violation?

Reply:
You're right, I have updated it as suggested.